dask/dot.py

```diff
@@ -273,13 +273,14 @@ def dot_graph(dsk, filename="mydask", format=None, **kwargs):
 
 def graphviz_to_file(g, filename, format):
     fmts = [".png", ".pdf", ".dot", ".svg", ".jpeg", ".jpg"]
+
+    if format is None and filename is None:
+        format = "png"
+
     if format is None and any(filename.lower().endswith(fmt) for fmt in fmts):
         filename, format = os.path.splitext(filename)
         format = format[1:].lower()
 
```

Comment:
Cat it happen that `format` is still `None` here, e.g., a filename was given but didn't match any of the known extensions?

Reply:
Oh yes! I shouldn't have deleted the previous check for that. This was the issue! The failed tests pass now :tada: :heavy_check_mark: